### ISSUE TITLE
use w3 color contrast algorithm

### ISSUE THREAD

**Issue by GoogleCodeExporter** [SEBDYA/googletransitdatafeed#108]
```
Could we change the contrast check for the feed validator and Transit to be
the w3c requirement? The w3c requirement is found here:
http://juicystudio.com/services/colourcontrast.php#contrast

Then the GTFS doc wouldn't have to explain contrast and could just point
people to existing resources, like this handy contrast picker.
http://snook.ca/technical/colour_contrast/colour.html

And, even better, we would have the warm happy joy of knowing that our
contrast colors for routes were fully accessible to viewers with vision
impairment. Everyone wins!
```

Original issue reported on code.google.com by `tom.brow...@gmail.com` on 6 Nov 2008 at 4:39

**Comment by GoogleCodeExporter** [comment]
```
The hardest part of this change is making the proposal to gtfs-changes.
```

Original comment by `tom.brow...@gmail.com` on 28 Oct 2009 at 10:57
- Added labels: **Difficulty-Easy**, **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
Original comment by `tom.brow...@gmail.com` on 11 Nov 2009 at 2:18
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
The current public GTFS spec (dated March 30, 2009) already says:

The color difference between route_color and route_text_color should provide
sufficient contrast when viewed on a black and white screen.
http://www.w3.org/TR/AERT#color-contrast offers a useful algorithm for 
evaluating
color contrast.

I will switch the luminance computation in the transitfeed code to the formula 
from
that doc but leave the threshold for a warning lower than their recommendation
(similar to where it was before), because they talk about running, normal-size 
text,
which has higher demands than line names that are not read at a longer stretch 
and
are probably shown in bigger font.

While doing this, I noticed that the spec's W3C reference is outdated.  (In 
fact,
it's a 9 year old draft(!).)  The current thing seems to be this:
http://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast-contrast  
Its
formula is more involved, and we should first discuss what we want in the spec 
before
writing code for it.

```

Original comment by `arnoegw.code@gmail.com` on 11 Nov 2009 at 5:37

**Comment by GoogleCodeExporter** [comment]
```
My fix announced above was reviewed by Tom at 
http://codereview.appspot.com/154054
and committed to SVN in revision 1145.  I'm keeping this issue open because of 
the
future work identified above.

```

Original comment by `arnoegw.code@gmail.com` on 11 Nov 2009 at 7:13

**Comment by GoogleCodeExporter** [comment]
```
Moved to https://github.com/google/transitfeed/issues/108
```

Original comment by `bdfer...@google.com` on 7 Oct 2014 at 7:59
- Changed state: **WontFix**
